compiler/rustc_data_structures/src/stable_hasher.rs

```diff
@@ -32,7 +32,7 @@ pub trait StableHasherResult: Sized {
 impl StableHasher {
     #[inline]
     pub fn new() -> Self {
-        StableHasher { state: SipHasher128::new_with_keys(0, 0) }
+        StableHasher { state: Hash128::with_seed(0) }
```

Comment:
Passing thought, should we perhaps consider using some other constant than 0 for the initial seed? Some hashers are known to produce pretty poor results when the initial seed is 0, which might not apply for these specific hash algorithms, but specifying a non-0 value would likely obviate concerns like these regardless.

Reply:
This should use `State::default()`, or a lot of redundant work will be done.